### PR TITLE
Update column grammar doc

### LINKED ISSUE #2282: Feature column doc improvement
SQLFlow supports data pre-processing using TensorFlow feature column APIs. The doc of feature column support can be seen in https://github.com/sql-machine-learning/sqlflow/blob/develop/doc/language_guide.md#feature-columns . But the doc is incomplete:

- `CATEGORY_HASH`, `INDICATOR` and `BUCKET` are supported in codes but not exposed in doc.
- `DENSE` and `SPARSE` are supported in codes but not exposed in doc.
- `CATEGORY_ID` supports vocabulary list in codes but not exposed in doc.
- The examples in doc are quite simple, and maybe we should have more examples. For example, write full SQL statements instead of part of them; add some examples to explain when we should use the certain feature column (considering the data type of the table column, etc).

### PR DESCRIPTION
**Pull request by typhoonzero**
Fix https://github.com/sql-machine-learning/sqlflow/issues/2282

### CODE REVIEW

File: doc/language_guide.md [hunk 1]
```diff
@@ -157,115 +157,101 @@ SQLFlow supports specifying various feature columns in the column clause and lab
...
+
+The format is: `CATEGORY_HASH([DENSE()|SPARSE()|field], BUCKET_SIZE)`, represents sparse feature where ids are set by hashing. In the expression:
+
+- `[DENSE()|SPARSE()|field]` you can specify the field data format using `DENSE`, `SPARSE` or just specify the column name and let SQLFlow to infer the actual format.
+- `BUCKET_SIZE` number of bucket of the hashed value.
+
+#### WEIGHTED_CATEGORY
+The format is: `WEIGHTED_CATEGORY([CATEGORY_ID(SPARSE(field,shape,delim,dtype_value,delim_kv,dtype_key))|CATEGORY_HASH(...)])`. Applies weight values to any categorical column like `CATEGORY_ID`. In the expression:
+
+- `WEIGHTED_CATEGORY` accepts one argument, and this argument must be an categorical column expression. The data format of the categorical column must be like `SPARSE(field,shape,delim,dtype_value,delim_kv,dtype_key)`, which will feed the categorical IDs and
```
Comment (lhw362950217) [inline doc/language_guide.md]: Is here missing something？
`categorical IDs and`

File: doc/language_guide.md [hunk 2]
```diff
@@ -157,115 +157,101 @@ SQLFlow supports specifying various feature columns in the column clause and lab
...
+
+- `[column_1, column_2]` a list of two elements respresenting two columns as the input.
+- `HASH_BUCKET_SIZE` number of buckets that the crossed feature will be hashed to.
+
+#### BUCKET
+
+The format is: `BUCKET([DENSE(...)|col_name], BOUNDARIES)`. Represents discretized dense input bucketed by `BOUNDARIES`. In the expression:
+
+- `[DENSE(...)|col_name]` is the input column, can be a numeric column with any shape.
+- `BOUNDARIES` is a list represents the value boundaries, like `[0., 1., 2.]` generates buckets `(-inf, 0.)`, `[0., 1.)`, `[1., 2.)`, and `[2., +inf)`.
```
Comment (lhw362950217) [inline doc/language_guide.md:268]: Maybe we can add some examples to illustrate the input and output for each column type.

Comment (typhoonzero) [reply doc/language_guide.md:268]: Input examples are in the above table, the users do not need to know the output format.

**Review by lhw362950217 — APPROVED** [review]
LGTM

**Pull request merged**